From a010d5d2339c553bc450b0e6fd81f1719f991bbb Mon Sep 17 00:00:00 2001
From: zglstudylinux <zglstudylinux@users.noreply.github.com>
Date: Fri, 16 Oct 2026 14:41:41 +0800
Subject: [PATCH 2/2] t113: zgl_board add RTL8723DS wifi packages, drop XR829

- modules.mk: new kmod-net-rtl8723ds package (8723ds.ko, deliberately
  no AUTOLOAD: BT firmware download must run first via rtk_hciattach,
  then rc.local loads 8723ds after hci0 is up)
- defconfig: USES_XR829 -> USES_REALTEK, enable rtl8723ds-firmware and
  kmod-net-rtl8723ds, disable xr829 firmware/module/rftest
---
 t113-common/modules.mk   | 16 ++++++++++++++++
 t113-zgl_board/defconfig | 12 ++++++------
 2 files changed, 22 insertions(+), 6 deletions(-)

diff --git a/t113-common/modules.mk b/t113-common/modules.mk
index 2e76194..559d18b 100755
--- a/t113-common/modules.mk
+++ b/t113-common/modules.mk
@@ -252,6 +252,22 @@ endef
 
 $(eval $(call KernelPackage,net-rtl8821cs))
 
+define KernelPackage/net-rtl8723ds
+  SUBMENU:=$(WIRELESS_MENU)
+  TITLE:=RTL8723DS support
+  DEPENDS:= +@IPV6 +@USES_REALTEK
+  FILES:=$(LINUX_DIR)/drivers/net/wireless/rtl8723ds/8723ds.ko
+  # No AUTOLOAD: the RTL8723DS WiFi SDIO firmware download would corrupt the
+  # module's BT ROM-download state. BT (rtk_hciattach) must run FIRST, then
+  # 8723ds is loaded explicitly by /etc/rc.local (base-files) once hci0 is up.
+endef
+
+define KernelPackage/net-rtl8723ds/description
+ Kernel modules for RealTek RTL8723DS support
+endef
+
+$(eval $(call KernelPackage,net-rtl8723ds))
+
 define KernelPackage/sunxi-disp
   SUBMENU:=$(VIDEO_MENU)
   TITLE:=sunxi-disp support
diff --git a/t113-zgl_board/defconfig b/t113-zgl_board/defconfig
index 2aa0f72..84f6a8f 100755
--- a/t113-zgl_board/defconfig
+++ b/t113-zgl_board/defconfig
@@ -3192,8 +3192,7 @@ CONFIG_PACKAGE_wirelesscommon=y
 # CONFIG_PACKAGE_bnx2-firmware is not set
 # CONFIG_PACKAGE_bnx2x-firmware is not set
 CONFIG_FIRMWARE_PATH="/lib/firmware/"
-# CONFIG_XR829_USE_40M_SDD is not set
-CONFIG_USES_XR829=y
+CONFIG_USES_REALTEK=y
 # CONFIG_PACKAGE_brcmfmac-firmware-4329-sdio is not set
 # CONFIG_PACKAGE_brcmfmac-firmware-43430-sdio-rpi-3b is not set
 # CONFIG_PACKAGE_brcmfmac-firmware-43430-sdio-rpi-zero-w is not set
@@ -3274,7 +3273,7 @@ CONFIG_USES_XR829=y
 # CONFIG_PACKAGE_rtl8192su-firmware is not set
 # CONFIG_PACKAGE_rtl8723au-firmware is not set
 # CONFIG_PACKAGE_rtl8723bu-firmware is not set
-# CONFIG_PACKAGE_rtl8723ds-firmware is not set
+CONFIG_PACKAGE_rtl8723ds-firmware=y
 # CONFIG_PACKAGE_rtl8733bs-firmware is not set
 # CONFIG_PACKAGE_rtl8821ae-firmware is not set
 # CONFIG_PACKAGE_rtl8821cs-firmware is not set
@@ -3291,7 +3290,7 @@ CONFIG_PACKAGE_wireless-regdb=y
 # CONFIG_PACKAGE_xr819-firmware is not set
 # CONFIG_PACKAGE_xr819a-firmware is not set
 # CONFIG_PACKAGE_xr819s-firmware is not set
-CONFIG_PACKAGE_xr829-firmware=y
+# CONFIG_PACKAGE_xr829-firmware is not set
 # end of Firmware
 
 #
@@ -4070,10 +4069,11 @@ CONFIG_PACKAGE_kmod-cfg80211=y
 # CONFIG_PACKAGE_kmod-mwifiex-sdio is not set
 # CONFIG_PACKAGE_kmod-net-broadcom is not set
 # CONFIG_PACKAGE_kmod-net-rtl8192su is not set
+CONFIG_PACKAGE_kmod-net-rtl8723ds=y
 # CONFIG_PACKAGE_kmod-net-rtl8821cs is not set
 # CONFIG_PACKAGE_kmod-net-xr819s is not set
 # CONFIG_PACKAGE_kmod-net-xr819s-40M is not set
-CONFIG_PACKAGE_kmod-net-xr829=y
+# CONFIG_PACKAGE_kmod-net-xr829 is not set
 # CONFIG_PACKAGE_kmod-net-xr829-40M is not set
 # CONFIG_PACKAGE_kmod-p54-common is not set
 # CONFIG_PACKAGE_kmod-p54-usb is not set
@@ -6982,7 +6982,7 @@ CONFIG_PACKAGE_e2fsprogs=y
 # CONFIG_PACKAGE_v821-rftest is not set
 # CONFIG_PACKAGE_xr819-rftest is not set
 # CONFIG_PACKAGE_xr819s-rftest is not set
-CONFIG_PACKAGE_xr829-rftest=y
+# CONFIG_PACKAGE_xr829-rftest is not set
 # end of rf test tool
 
 #
-- 
2.43.0

